### PR TITLE
feat: add some missing APIs to the API reference and feature matrix

### PR DESCRIPTION
**Pull request by cprice404**
This commit adds the following missing APIs to the reference and
feature matrix:

* itemGetType
* itemGetTtl
* setLength
* dictionaryLength
* sortedSetLength
* sortedSetLengthByScore
* generateAuthToken
* refreshAuthToken

### CODE REVIEW

File: docs/develop/api-reference/sorted-set-collections.md [hunk 1]
```diff
@@ -271,3 +271,45 @@ Example:
...
+</details>
+
+### SortedSetLengthByScore
+For an existing sorted set item, finds all of the values between the specified min and max score and 
+
+| Name           | Type         | Description                                |
+|----------------| ------------ |--------------------------------------------|
+| cacheName      | String       | Name of the cache.                         |
+| sortedSetName | String       | Name of the sorted set item to be checked. |
+| minScore | Optional[double]   | The inclusive low score of the results. Default is `-inf`, ie include through the lowest score. |
```
Comment (nidhidham) [inline docs/develop/api-reference/sorted-set-collections.md:302]: Are we finalizing this?

**Review by nidhidham — DISMISSED** [review]
setLength, dictionaryLength, sortedSetLength, sortedSetLengthByScore - LGTM :)

**Review by honeyAcorn — DISMISSED** [review]
LGTM for `itemGetType` and `itemGetTtl`

File: docs/develop/api-reference/sorted-set-collections.md (hunk 1 above)
Comment (cprice404) [reply docs/develop/api-reference/sorted-set-collections.md:302]: @allenheltondev are you cool with this for now?  I think this is what it needs to look like here, in order to match existing SortedSet functions.  I can sync w/you on the final shape of the JS SDK API when we add it there.

File: docs/develop/api-reference/dictionary-collections.md [hunk 2]
```diff
@@ -208,4 +208,25 @@ Sets several field:value pairs in a dictionary item. If the dictionary item does
 
 See [response objects](./response-objects.md) for specific information.
 
-</details>
\ No newline at end of file
+</details>
+
+### DictionaryLength
+Get the length of an existing dictionary item
```
Comment (allenheltondev) [inline docs/develop/api-reference/dictionary-collections.md:214]: This is a bit ambiguous to me. I'd reword to "Get the number of entries in a dictionary item"

File: docs/develop/api-reference/index.mdx [hunk 3]
```diff
@@ -137,6 +137,29 @@ Sends a ping to the server. This API can be used for checking connectivity to co
...
+
+| Name            | Type            | Description                                   |
+| --------------- | --------------- | --------------------------------------------- |
+| cacheName       | String          | Name of the cache.                            |
+| key        | String \| Byte          | Key whose item type should be returned.              |
+
+<details>
+  <summary>Method response object</summary>
+
+  * Cache hit
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:152]: These are formatted inconsistently with the dictionary responses. The dictionary ones have 
* Hit
* Miss 
* Error

We should be consistent with the return types. Personally, I prefer hit, miss, error over cache hit, cache miss, and cache error

File: docs/develop/api-reference/index.mdx [hunk 4]
```diff
@@ -241,7 +264,81 @@ Examples
...
 
-### Collection data types (CDTs)
+### ItemGetTtl
+
+For a given key, returns the duration of time remaining (Time To Live) before the item expires from the cache.
+
+| Name            | Type            | Description                                   |
+| --------------- | --------------- | --------------------------------------------- |
+| cacheName       | String          | Name of the cache.                            |
+| key        | String \| Byte          | Key whose item type should be returned.              |
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:274]: The description is wrong for this property

File: docs/develop/api-reference/dictionary-collections.md [hunk 5]
```diff
@@ -208,4 +208,25 @@ Sets several field:value pairs in a dictionary item. If the dictionary item does
...
+
+| Name           | Type         | Description                                |
+|----------------| ------------ |--------------------------------------------|
+| cacheName      | String       | Name of the cache.                         |
+| dictionaryName | String       | Name of the dictionary item to be checked. |
```
Comment (allenheltondev) [inline docs/develop/api-reference/dictionary-collections.md:219]: This is inconsistent with the other APIs. Here we have `dictionaryName` but in the get type and get ttl commands we just have `key`

File: docs/develop/api-reference/index.mdx [hunk 6]
```diff
@@ -241,7 +264,81 @@ Examples
...
+
+</details>
+
+## Auth APIs
+
+These APIs are used to manage Momento auth tokens and access.
+
+### GenerateAuthToken
+
+Generates a new Momento Auth token with the specified permissions and expiry.
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:295]: Auth should be lowercase here

File: docs/develop/api-reference/index.mdx [hunk 7]
```diff
@@ -241,7 +264,81 @@ Examples
...
+
+| Name            | Type            | Description                                   |
+| --------------- | --------------- | --------------------------------------------- |
+| scope           | TokenScope      | The permissions to grant to the new token.  Pre-built TokenScope objects are provided by the SDKs. |
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:299]: We should list somewhere in the docs what the token scopes are and what permissions they allow

File: docs/develop/api-reference/index.mdx [hunk 8]
```diff
@@ -241,7 +264,81 @@ Examples
...
+| expiresIn       | ExpiresIn       | The number of seconds until the token expires, or `never`.              |
+
+<details>
+  <summary>Method response object</summary>
+
+  * Success
+  - `authToken`: string - the new auth token
+  - `refreshToken`: string - a refresh token that can be used with the `RefreshAuthToken` API to refresh the token before it expires
+  - `expiresAt`: Timestamp - the timestamp at which the token will expire
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:308]: This is an object, isn't it?

File: docs/develop/api-reference/set-collections.md [hunk 9]
```diff
@@ -164,3 +164,23 @@ The response object for SetContainsElements returns three possible options, a ca
...
 
 </details>
+
+### SetLength
+Get the length of an existing set item
+
+| Name      | Type         | Description                         |
+|-----------| ------------ |-------------------------------------|
+| cacheName | String       | Name of the cache.                  |
+| setName   | String       | Name of the set item to be checked. |
```
Comment (allenheltondev) [inline docs/develop/api-reference/set-collections.md:174]: Are these key properties strings or strings/byte array? Are all our commands the same? I've seen both in this PR

File: docs/develop/api-reference/sorted-set-collections.md [hunk 10]
```diff
@@ -271,3 +271,45 @@ Example:
 | Score           | Signed double 64-bit float   | Score the element. |
 
 A SortedSetElement can exist by itself or as part of an array of SortedSetElements
+
+### SortedSetLength
+Get the length of an existing sorted set item
```
Comment (allenheltondev) [inline docs/develop/api-reference/sorted-set-collections.md]: Similar to the dictionary, I'd reword this to "Get the number of entries in a sorted set cache item"

File: docs/develop/api-reference/sorted-set-collections.md (hunk 1 above)
Comment (allenheltondev) [reply docs/develop/api-reference/sorted-set-collections.md:302]: Yeah this is fine, but it leaves me with an existential crisis about these docs in general. If it's different for all our SDKs and end users don't call these APIs directly - then what is this documenting? The back-end that users don't see?

Comment (cprice404) [reply docs/develop/api-reference/sorted-set-collections.md:302]: @allenheltondev I will be adding code snippets for all the languages for all of the APIs very soon.  The purpose of this PR for me is to get these in place so i have a place to add the snippet react tags, so I can then have my team start filling in the code samples.

File: docs/develop/api-reference/sorted-set-collections.md [hunk 11]
```diff
@@ -271,3 +271,45 @@ Example:
...
+    * `length()`: Number
+* Miss
+* Error
+
+See [response objects](./response-objects.md) for specific information.
+
+</details>
+
+### SortedSetLengthByScore
+For an existing sorted set item, finds all of the values between the specified min and max score and
```
Comment (NoSQLKnowHow) [inline docs/develop/api-reference/sorted-set-collections.md:296]: This sentence looks unfinished.

File: docs/develop/api-reference/sorted-set-collections.md [hunk 12]
```diff
@@ -270,4 +270,46 @@ Example:
...
```
Comment (NoSQLKnowHow) [inline docs/develop/api-reference/sorted-set-collections.md]:
```suggestion
For an existing sorted set item, it finds all of the values between the specified min and max score and returns the length.
```

**Review by NoSQLKnowHow — APPROVED** [review]
